index.php

```diff
@@ -36,7 +36,8 @@ function bootstrap() {
 
 	$cache = new MenuCache();
 
-	add_filter( 'pre_wp_nav_menu', [ $cache, 'get_menu' ], 10, 2 );
+	// Run as early as possible. All other code here would either be a conflicting cache plugin or a misplaced attempt to initialize a menu with no way to restore after we've run. ( switch_to_blog() on this hook would be fatal )
+	add_filter( 'pre_wp_nav_menu', [ $cache, 'get_menu' ], 0, 2 );
```

Comment:
Technically, `0` is not "as early as possible", but it is fine as it ist, I guess.
Since `0` might be an obvious choice for a lot of people who want to run _as early as possible_, using `-1` here might be even better, though. What do you think?